src/parser/warrior/protection/modules/azerite/BraceForImpact.js

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+
+import { calculateAzeriteEffects } from 'common/stats';
+import { formatNumber, formatMilliseconds } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import SpellLink from 'common/SpellLink';
+import HIT_TYPES from 'game/HIT_TYPES';
+
+import StatTracker from 'parser/shared/modules/StatTracker';
+import Analyzer from 'parser/core/Analyzer';
+import Combatants from 'parser/shared/modules/Combatants';
+import AzeritePowerStatistic from 'interface/statistics/AzeritePowerStatistic';
+
+class BraceForImpact extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+    abilityTracker: AbilityTracker,
+  };
+
+  traitBlock = 0;
+  lastApplication = 0;
+  currentStack = 0;
+  mappy;
+  shieldBlockBuff = this.selectedCombatant.hasTalent(SPELLS.HEAVY_REPERCUSSIONS_TALENT.id) ? 1.6 : 1.3;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.BRACE_FOR_IMPACT.id);
+    if (!this.active) {
+      return;
+    }
+    const ranks = this.selectedCombatant.traitRanks(SPELLS.BRACE_FOR_IMPACT.id) || [];
+    this.traitDamage = ranks.reduce((total, rank) => total + calculateAzeriteEffects(SPELLS.BRACE_FOR_IMPACT.id, rank)[1], 0);
+    this.mappy = new Map();
+    this.lastApplication = this.owner.fight.start_time;
+  }
+
+  on_byPlayer_damage(event) {
```

Comment:
I am aware that calculateAzeriteDamage doesn't work correctly for melee spells since attack power coefficients work different because of weapon damage, but if you do want to make a calculation that doesn't account for unknown damage increasing effects, you should at least make that clear to the user.